ganttproject/src/net/sourceforge/ganttproject/gui/AbstractTableAndActionsComponent.java

```diff
@@ -218,6 +218,30 @@ public static JPanel createDefaultTableAndActions(JComponent table, JComponent a
     return result;
   }
 
+  public static <T> JPanel createDefaultTableAndActions(JTable table, TableModelExt modelExt) {
+    AbstractTableAndActionsComponent<T> component = new AbstractTableAndActionsComponent<T>(table) {
+      @Override
+      protected void onAddEvent() {
+        getTable().editCellAt(modelExt.getRowCount() - 1, 1);
+      }
+
+      @Override
+      protected void onDeleteEvent() {
+        if (getTable().isEditing() && getTable().getCellEditor() != null) {
+          getTable().getCellEditor().stopCellEditing();
+        }
+        modelExt.delete(getTable().getSelectedRows());
+      }
+
+      @Override
+      protected T getValue(int row) {
+        java.util.List<T> values = modelExt.getAllValues();
```

Comment:
I think you can use short name `List` here. It is used a few times in this file.